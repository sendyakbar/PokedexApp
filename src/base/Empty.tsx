import React from 'react';
import {StyleSheet, View, Text, ViewStyle, TextStyle} from 'react-native';

export default function Empty(): JSX.Element {
  return (
    <View style={styles.container}>
      <Text style={styles.emptyText}>Not found...</Text>
    </View>
  );
}

type StyleType = {
  container: ViewStyle;
  emptyText: TextStyle;
};

const styles = StyleSheet.create<StyleType>({
  container: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
  },
  emptyText: {
    fontSize: 12,
    color: 'grey',
    fontStyle: 'italic',
  },
});
