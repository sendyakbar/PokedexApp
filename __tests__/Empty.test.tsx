import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import Empty from '../src/base/Empty';

it('renders correctly', () => {
  const tree = renderer.create(<Empty />).toJSON();
  expect(tree).toMatchSnapshot();
});
