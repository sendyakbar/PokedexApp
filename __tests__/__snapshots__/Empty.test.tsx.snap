// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "grey",
        "fontSize": 12,
        "fontStyle": "italic",
      }
    }
  >
    Not found...
  </Text>
</View>
`;
